azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_servicefabric.go

```diff
@@ -0,0 +1,216 @@
+			Parameters: &parameters,
+			Scheme:     converter.String("UsernamePassword"),
+		}
+		return serviceEndpoint, projectID, nil
+	}
+
+	return nil, nil, fmt.Errorf("One of %s or %s blocks must be specified", resourceBlockServiceFabricAzureActiveDirectory, resourceBlockServiceFabricCertificate)
+}
+
+func expandServiceEndpointServiceFabricServerCertificateLookup(configuration map[string]interface{}) map[string]string {
+	certLookup := configuration["server_certificate_lookup"].(string)
+	parameters := map[string]string{
+		"certLookup": certLookup,
+	}
+	switch certLookup {
+	case "Thumbprint":
+		parameters["servercertthumbprint"] = configuration["server_certificate_thumbprint"].(string)
+	case "CommonName":
+		parameters["servercertcommonname"] = configuration["server_certificate_common_name"].(string)
+	}
+	return parameters
+}
+
+func flattenServiceFabricCertificate(serviceEndpoint *serviceendpoint.ServiceEndpoint, hashKeyClientCertificate string, hashValueClientCertificate string, hashKeyClientCertificatePassword string, hashValueClientCertificatePassword string) interface{} {
+	result := flattenServiceEndpointServiceFabricServerCertificateLookup(serviceEndpoint)
+	result[0]["client_certificate"] = (*serviceEndpoint.Authorization.Parameters)["certificate"]
+	result[0]["client_certificate_password"] = (*serviceEndpoint.Authorization.Parameters)["certificatepassword"]
+	result[0][hashKeyClientCertificate] = hashValueClientCertificate
+	result[0][hashKeyClientCertificatePassword] = hashValueClientCertificatePassword
+	return result
+}
+
+func flattenServiceFabricAzureActiveDirectory(serviceEndpoint *serviceendpoint.ServiceEndpoint, hashKeyPassword string, hashValuePassword string) interface{} {
+	result := flattenServiceEndpointServiceFabricServerCertificateLookup(serviceEndpoint)
+	result[0]["username"] = (*serviceEndpoint.Authorization.Parameters)["username"]
+	result[0]["password"] = (*serviceEndpoint.Authorization.Parameters)["password"]
+	result[0][hashKeyPassword] = hashValuePassword
+	return result
+}
+
+func flattenServiceEndpointServiceFabricServerCertificateLookup(serviceEndpoint *serviceendpoint.ServiceEndpoint) []map[string]interface{} {
+	certLookup := (*serviceEndpoint.Authorization.Parameters)["certLookup"]
+	result := []map[string]interface{}{{
+		"server_certificate_lookup": certLookup,
+	}}
+	switch certLookup {
+	case "Thumbprint":
+		result[0]["server_certificate_thumbprint"] = (*serviceEndpoint.Authorization.Parameters)["servercertthumbprint"]
+	case "CommonName":
+		result[0]["server_certificate_common_name"] = (*serviceEndpoint.Authorization.Parameters)["servercertcommonname"]
+	}
+	return result
+}
+
+// Convert AzDO data structure to internal Terraform data structure
+func flattenServiceEndpointServiceFabric(d *schema.ResourceData, serviceEndpoint *serviceendpoint.ServiceEndpoint, projectID *string) {
+	doBaseFlattening(d, serviceEndpoint, projectID)
+
+	switch *serviceEndpoint.Authorization.Scheme {
```

Comment:
Service fabric support three type schema: `Certificate`, `UsernamePassword` and `None`. Current resource does not support `None`. If someone import an service fabric create in schema `None`,  provider should  notify user that this resource was not supported.